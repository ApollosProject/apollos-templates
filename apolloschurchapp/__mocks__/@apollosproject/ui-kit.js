module.exports = {
  ...jest.requireActual('@apollosproject/ui-kit'),
  NavigationService: {
    navigate: jest.fn(),
    setTopLevelNavigator: jest.fn(),
  },
};
