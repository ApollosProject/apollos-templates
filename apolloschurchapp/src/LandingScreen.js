import React from 'react';
import { styled, GradientOverlayImage } from '@apollosproject/ui-kit';

import ApollosLandingScreen from './ui/LandingScreen';

const FullScreenImage = styled({
  resizeMode: 'cover',
  position: 'absolute',
})(GradientOverlayImage);

const LandingScreen = ({ navigation }) => (
  <ApollosLandingScreen
    onPressPrimary={() => navigation.push('Auth')}
    textColor={'white'}
    BackgroundComponent={
      <FullScreenImage source={'https://picsum.photos/375/812/?random'} />
    }
    primaryNavText={"Let's go!"}
  />
);

LandingScreen.navigationOptions = {
  header: null,
};

export default LandingScreen;
