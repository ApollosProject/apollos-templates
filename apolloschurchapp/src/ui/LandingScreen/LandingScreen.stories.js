import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { GradientOverlayImage } from '@apollosproject/ui-kit';

import LandingScreen from '.';

storiesOf('LandingScreen', module)
  .add('default', () => <LandingScreen />)
  .add('slideTitle', () => <LandingScreen slideTitle={'Custom title text'} />)
  .add('description', () => (
    <LandingScreen description={'Custom description text'} />
  ))
  .add('textColor', () => <LandingScreen textColor={'salmon'} />)
  .add('ImageComponent', () => (
    <LandingScreen
      BackgroundComponent={
        <GradientOverlayImage
          source={'https://picsum.photos/375/812/?random'}
        />
      }
    />
  ))
  .add('Slide props', () => <LandingScreen onPressPrimary={() => {}} />);
