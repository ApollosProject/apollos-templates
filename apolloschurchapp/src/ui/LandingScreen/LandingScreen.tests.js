import React from 'react';
import renderer from 'react-test-renderer';
import { GradientOverlayImage } from '@apollosproject/ui-kit';

import Providers from '../../Providers';

import LandingScreen from '.';

describe('The Onboarding LandingScreen component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom title', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen slideTitle={'Custom title text'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom description', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen description={'Custom description text'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom text color', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen textColor={'salmon'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with an ImageComponent', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen
          BackgroundComponent={
            <GradientOverlayImage
              source={'https://picsum.photos/375/812/?random'}
            />
          }
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should pass additional props to Slide', () => {
    const tree = renderer.create(
      <Providers>
        <LandingScreen onPressPrimary={jest.fn()} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
