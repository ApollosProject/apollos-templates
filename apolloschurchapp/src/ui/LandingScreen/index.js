import React from 'react';
import { StyleSheet, Image } from 'react-native';
import { styled, withTheme } from '@apollosproject/ui-kit';

import LandingScreen from './LandingScreen';

const FullScreenImage = styled({
  resizeMode: 'cover',
  ...StyleSheet.absoluteFill,
  width: '100%',
  height: '100%',
})(Image);

const ThemedLandingScreen = withTheme(({ theme }) => ({
  textColor: theme.colors.primary,
}))(LandingScreen);

const LandingScreenSlide = ({ navigation }) => (
  <ThemedLandingScreen
    onPressPrimary={() => navigation.push('Auth')}
    BackgroundComponent={
      <FullScreenImage source={require('./img/landing.jpg')} />
    }
    primaryNavText={"Let's go!"}
  />
);

export default LandingScreenSlide;
