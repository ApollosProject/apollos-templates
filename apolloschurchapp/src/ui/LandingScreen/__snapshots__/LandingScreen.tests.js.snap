// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding LandingScreen component should pass additional props to Slide 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            We're glad you're here.
          </Text>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            We're not just a building you go to, but a family to belong to.
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row-reverse",
          "justifyContent": "space-between",
          "left": 0,
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "paddingVertical": 0,
          "position": "absolute",
          "right": 0,
        }
      }
      vertical={false}
    >
      <RCTSafeAreaView
        edges={
          Array [
            "bottom",
          ]
        }
        emulateUnlessSupported={true}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            bordered={false}
            disabled={false}
            pill={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00676D",
                "borderColor": "#00676D",
                "borderRadius": 48,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                }
              }
            >
              Next
            </Text>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "marginLeft": 8,
                    "marginRight": -8,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={20}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4278190080,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGPath
                  d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
                  fill={
                    Array [
                      0,
                      4294967295,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </BVLinearGradient>,
]
`;

exports[`The Onboarding LandingScreen component should render 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            We're glad you're here.
          </Text>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            We're not just a building you go to, but a family to belong to.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </BVLinearGradient>,
]
`;

exports[`The Onboarding LandingScreen component should render a custom description 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            We're glad you're here.
          </Text>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            Custom description text
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </BVLinearGradient>,
]
`;

exports[`The Onboarding LandingScreen component should render a custom title 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            Custom title text
          </Text>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            We're not just a building you go to, but a family to belong to.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </BVLinearGradient>,
]
`;

exports[`The Onboarding LandingScreen component should render with a custom text color 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            color={4294606962}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            tintColor={4294606962}
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4294606962,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            color="salmon"
            style={
              Object {
                "color": "salmon",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            We're glad you're here.
          </Text>
          <Text
            color="salmon"
            style={
              Object {
                "color": "salmon",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            We're not just a building you go to, but a family to belong to.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </BVLinearGradient>,
]
`;

exports[`The Onboarding LandingScreen component should render with an ImageComponent 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      scrollEnabled={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#A5A5A5",
              "overflow": "hidden",
              "width": "100%",
            }
          }
        >
          <Image
            fadeDuration={250}
            onLoad={[Function]}
            source={
              Array [
                Object {
                  "cache": "force-cache",
                  "uri": "https://picsum.photos/375/812/?random",
                  "url": "https://picsum.photos/375/812/?random",
                },
              ]
            }
            style={
              Object {
                "aspectRatio": 1,
                "backgroundColor": "#A5A5A5",
                "resizeMode": "cover",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={48}
            bbWidth={48}
            focusable={false}
            height={48}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 16,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 48,
                  "width": 48,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={48}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "marginBottom": 16,
              }
            }
          >
            We're glad you're here.
          </Text>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            We're not just a building you go to, but a family to belong to.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </BVLinearGradient>,
]
`;
